### ISSUE TITLE
Can't specify ssh_username via machine_options

### ISSUE THREAD

**Issue by miguelcnf** [chef/chef-provisioning-aws#44]
Because of:

```ruby
          machine_spec.location = {
              'driver_url' => driver_url,
              'driver_version' => Chef::Provisioning::AWSDriver::VERSION,
              'allocated_at' => Time.now.utc.to_s,
              'host_node' => action_handler.host_node,
              'image_id' => machine_options[:image_id],
              'instance_id' => instance.id
          }
```

And:

```ruby
          username = machine_spec.location['ssh_username'] || default_ssh_username
```

We can't set ssh_username as a machine_option.

**Comment by jkeiser** [comment]
This should be fixed with the merge of #65, would you mind retesting when you get a chance?

**Comment by jkeiser** [comment]
I'm going to close in the meantime, but @miguelcnf please do reopen if #65 didn't fix it--I know you had a different fix in the offing.
